yarn.config.cjs

```diff
@@ -488,20 +490,35 @@ async function expectWorkspaceLicense(workspace) {
 function expectCorrectWorkspaceExports(workspace) {
   // All non-root packages must provide the location of the ESM-compatible
   // JavaScript entrypoint and its matching type declaration file.
-  expectWorkspaceField(workspace, 'exports["."].import', './dist/index.mjs');
   expectWorkspaceField(
     workspace,
-    'exports["."].types',
-    './dist/types/index.d.ts',
+    'exports["."].import.types',
+    './dist/index.d.mts',
+  );
+  expectWorkspaceField(
+    workspace,
+    'exports["."].import.default',
+    './dist/index.mjs',
   );
-  // TODO: This was copied from the module template: enable when ready
-  // expectWorkspaceField(workspace, 'module', './dist/index.mjs');
 
   // All non-root package must provide the location of the CommonJS-compatible
   // entrypoint and its matching type declaration file.
-  expectWorkspaceField(workspace, 'exports["."].require', './dist/index.js');
-  expectWorkspaceField(workspace, 'main', './dist/index.js');
-  expectWorkspaceField(workspace, 'types', './dist/types/index.d.ts');
+  expectWorkspaceField(
+    workspace,
+    'exports["."].require.types',
+    './dist/index.d.cts',
+  );
+  expectWorkspaceField(
+    workspace,
+    'exports["."].require.default',
+    './dist/index.cjs',
+  );
+  expectWorkspaceField(workspace, 'main', './dist/index.cjs');
+  expectWorkspaceField(workspace, 'types', './dist/index.d.cts');
+
+  // Types should not be set in the export object directly, but rather in the
+  // `import` and `require` subfields.
+  expectWorkspaceField(workspace, 'exports["."].types', null);
```

Comment:
Just to confirm, would requiring `'exports["."].{require,import}.{types,default}'` also prohibit cases like these?

```json
{
  "exports": {
    ".": "./index.js"
  }
} 
```

```json
{
  "exports": "./index.js"
} 
```

Reply:
Yes. I don't think this is something we want to support anyway.

Reply:
I was wondering if we might need to forbid edge cases like these explicitly, but sounds like that won't be necessary.